pkg/webhook/resources/virtualmachine/validator.go

```diff
@@ -407,6 +409,21 @@ func (v *vmValidator) checkResizeVolumes(oldVM, newVM *kubevirtv1.VirtualMachine
 		}
 	}
 
+	for _, oldVMpvc := range oldPvcs {
+		for _, newVMpvc := range newPvcs {
+			if oldVMpvc.Name == newVMpvc.Name {
+				if *newVMpvc.Spec.StorageClassName != *oldVMpvc.Spec.StorageClassName {
+					return fmt.Errorf("storage class %v of volume %v in annotation %v cannot be changed to %v",
```

Comment:
You exist after the first error, wouldn't it be better to use an array of error, to collect all potential issues and exist if the array is greater than 0?

Reply:
This is very good suggestion, the only downside I can think of is if you have multiple changes to storageclassname webhook error might get a big message. Otherwise I agree that it can be a better UX to get all errors in 1 shot as opposed to:
1. Change a,b,c storage class names
2. Apply; Get rejected on `a`; Fix `a`
3. Apply; Get rejected on `b`; Fix `b`
...

wdyt @ihcsim @w13915984028 do we have some guideline on what the webhook size/content should be?

If not I can go ahead and refactor

Reply:
Ok, will stick with the standard for now, and will leave the discussion open if someone insists on certain format

Reply:
ok, I can try building a string so final result can be shorter:
```
storage class names for the volumes in the harvesterhci.io/volumeClaimTemplates annotation cannot be changed; name %s in volume %s;name %s in volume %s; name %s in volume %s;...
```
I think we that format can hit both all issues and smaller message, will update PR shortly